File: frontend/src/app/service/order/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Order } from '../../component/order/order';
import { Observable } from 'rxjs';

const BASE_URL = ["http://localhost:8080/api/manager"]

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  fetchAllOrders():Observable<Order[]> {
    return this.http.get<Order[]>(BASE_URL + `/order`);
  }

  createOrder(data: Order) {
    return this.http.post<Order>(BASE_URL + `/order`, data);
  }

  updateOrder(data: Order) {
    return this.http.put<Order>(BASE_URL + `/order/${data.id_order}`, data);
  }

  deleteOrder(id_order: Number) {
    return this.http.delete<Order>(BASE_URL + `/order/${id_order}`);
  }
}

File: backend/src/main/java/com/g0urd/grad_project/service/OrderService/OrderService.java
package com.g0urd.grad_project.service.OrderService;

import java.util.List;

import com.g0urd.grad_project.models.Order;

public interface OrderService {

    List<Order> fetchAllOrders();

    Order findById(Long id);

    Order createOrder(Order order);

    Order updateOrder(Order order);

    Boolean deleteOrder(Long id);

}
